src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -832,7 +833,9 @@ protected virtual bool RequiresPagedDataForParamCore(TexlNode[] args, int paramI
         public bool RequiresPagedDataForParam(CallNode callNode, int paramIndex, TexlBinding binding)
         {
             Contracts.AssertValue(callNode);
-            Contracts.AssertValue(callNode.Args);
+
+            // ConcatenateFunction is a special case where we may not have any args on call node.
+            Contracts.Assert(this is ConcatenateFunction || callNode.Args.Count > 0);

```

Comment:
if Concatenate can have no Args, 
wont this `callNode.Args.Children[paramIndex]` still fail ?

Reply:
Oops, it should have args because I am including them now from my change in Binder